examples/tutorials/mvdr_tutorial.py

```diff
@@ -38,258 +52,315 @@
+
+spectrum_mix = stft(waveform_mix)
+spectrum_clean = stft(waveform_clean)
+spectrum_noise = stft(waveform_noise)
 
-spec_mix = stft(mix)
-spec_clean = stft(clean)
-spec_reverb_clean = stft(reverb_clean)
-spec_noise = stft(noise)
 
 ######################################################################
-# Generate the Ideal Ratio Mask (IRM)
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-#
-# .. note::
-#    We found using the mask directly peforms better than using the
-#    square root of it. This is slightly different from the definition of IRM.
+# 3.2.1. Visualize mixture speech
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
 
-
-def get_irms(spec_clean, spec_noise):
-    mag_clean = spec_clean.abs() ** 2
-    mag_noise = spec_noise.abs() ** 2
-    irm_speech = mag_clean / (mag_clean + mag_noise)
-    irm_noise = mag_noise / (mag_clean + mag_noise)
-
-    return irm_speech, irm_noise
+plot_specgram(spectrum_mix[0], "Spectrogram of Mixture Speech (dB)")
+Audio(waveform_mix[0], rate=SAMPLE_RATE)
 
 
 ######################################################################
-# .. note::
-#    We use reverberant clean speech as the target here,
-#    you can also set it to dry clean speech.
+# 3.2.2. Visualize clean speech
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
 
-irm_speech, irm_noise = get_irms(spec_reverb_clean, spec_noise)
+plot_specgram(spectrum_clean[0], "Spectrogram of Clean Speech (dB)")
+Audio(waveform_clean[0], rate=SAMPLE_RATE)
 
-######################################################################
-# Apply MVDR
-# ----------
-#
 
 ######################################################################
-# Apply MVDR beamforming by using multi-channel masks
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 3.2.3. Visualize noise
+# ^^^^^^^^^^^^^^^^^^^^^^
 #
 
-results_multi = {}
-for solution in ["ref_channel", "stv_evd", "stv_power"]:
-    mvdr = torchaudio.transforms.MVDR(ref_channel=0, solution=solution, multi_mask=True)
-    stft_est = mvdr(spec_mix, irm_speech, irm_noise)
-    est = istft(stft_est, length=mix.shape[-1])
-    results_multi[solution] = est
+plot_specgram(spectrum_noise[0], "Spectrogram of Noise (dB)")
+Audio(waveform_noise[0], rate=SAMPLE_RATE)
+
 
 ######################################################################
-# Apply MVDR beamforming by using single-channel masks
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 3.3. Define the reference microphone
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We choose the first microphone in the array as the reference channel for demonstration.
+# The selection of the reference channel may depend on the design of the microphone array.
 #
-# We use the 1st channel as an example.
-# The channel selection may depend on the design of the microphone array
+# You can also apply a neural network to estimate the reference channel and
+# pass it to the MVDR module in an end-to-end speech enhancement model.
```

Comment:
i feel like this could be clearer. are we suggesting that such an "end-to-end speech enhancement model" would comprise a neural network that estimates the reference channel plus the MVDR module? or is only the MVDR module part of the end-to-end speech enhancement model?